engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -131,9 +133,13 @@ public WorldRendererLwjgl(WorldProvider worldProvider, ChunkProvider chunkProvid
         }
         localPlayerSystem.setPlayerCamera(playerCamera);
 
+        initMainDirectionalLight();
+
         renderableWorld = new RenderableWorldImpl(worldProvider, chunkProvider, bufferPool, playerCamera, shadowMapCamera);
         renderQueues = renderableWorld.getRenderQueues();
+    }
 
+    private void initMainDirectionalLight() {
         mainDirectionalLight.lightType = LightComponent.LightType.DIRECTIONAL;
```

Comment:
Hmm... I'ld probably argue that:
* WorldRenderer should not be setting up, maintaining or storing lighting state, that should all be done external to it.
* WorldRenderer should be finding and rendering lights from the entity system, including directional lights.

Or in essence best case would be the light component being on a celestial body entity (sun, moon, etc) and a separate system creating (if not using auto-entities) and updating those.

Otherwise I don't hate the reuse of the component - better than duplicating code.